app/instrumentation-tests/src/org/commcare/androidTests/ConditionalEnumInCaseListTests.kt

```diff
@@ -0,0 +1,162 @@
+import org.commcare.dalvik.R
+import org.commcare.utils.InstrumentationUtility
+
+import org.commcare.utils.isPresent
+import org.hamcrest.Matchers
+import org.junit.After
+
+import org.junit.Assert.*
+import org.junit.Before
+import org.junit.FixMethodOrder
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.MethodSorters
+import java.time.LocalDate
+import java.util.*
+
+
+@RunWith(AndroidJUnit4::class)
+@LargeTest
+@BrowserstackTests
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)  //executes the tests in order
+class ConditionalEnumInCaseListTests: BaseTest() {
+    companion object {
+        const val CCZ_NAME = "conditional_enum_in_case_list.ccz"
+        const val APP_NAME = "Conditional Enum in Case List Tests"
+        val date = LocalDate.now().toString()
+        //the list contains the form input values of name, age, appointment date and appointment attended
+        val listOfCases = listOf(
+            listOf("Test One "+date, "16", "Today", "yes"),
+            listOf("Test Two "+date, "21", "Yesterday", "no")
+        )
+    }
+
+    @Before
+    fun setup() {
+        installApp(APP_NAME, CCZ_NAME)
+        InstrumentationUtility.login("test1", "123")
+    }
+
+    /**
+     * Adding the teardown method so the tests doesnot fail after only thefirst execution, when executed as a whole.
+     */
+    @After
+    fun tearDown(){
+        InstrumentationUtility.logout()
+        InstrumentationUtility.uninstallCurrentApp()
+    }
+    @Test
+    fun testCreateCases(){
+        for (caseList in listOfCases) {
+            InstrumentationUtility.openForm(0,0)
+            onView(ViewMatchers.withClassName(Matchers.endsWith("EditText")))
+                .perform(ViewActions.typeText((caseList[0])))
+            InstrumentationUtility.nextPage()
+            if (caseList[2] == "Yesterday") {
+                setDateTo(-1)
+                Thread.sleep(1000) // without the wait the test fails with AmbiguousViewMatcherException and does not click the next button
+
+            } else if (caseList[2] == "Today") {
+                setDateTo(0)
+            }
+            InstrumentationUtility.nextPage()
+            onView(ViewMatchers.withClassName(Matchers.endsWith("EditText")))
+                .perform(ViewActions.typeText((caseList[1])))
+            InstrumentationUtility.submitForm()
+            assertTrue(onView(ViewMatchers.withText("1 form sent to server!")).isPresent())
+        }
+    }
+
+    @Test
+    fun testUpdateCases(){
+        for (caseList in listOfCases) {
+            InstrumentationUtility.openForm(0,1)
+            onView(ViewMatchers.withId(R.id.search_action_bar)).perform(ViewActions.click())
+            onView(ViewMatchers.withId(R.id.search_src_text)).perform(ViewActions.typeText(caseList[0]))
+            onView(ViewMatchers.withId(R.id.screen_entity_detail_list)).isPresent()
+            InstrumentationUtility.clickListItem(R.id.screen_entity_select_list, 0)
```

Comment:
these 4 lines can be abstracted as `InstrumentationUtility.searchAndEnterCase()` and can be resued in verification as well.

Reply:
created the same.